flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/source/HBaseSourceITCase.java

```diff
@@ -0,0 +1,313 @@
+        cluster.makeTable(baseTableName, DEFAULT_CF_COUNT);
+        String[] expectedValues = uniqueValues(2 * DEFAULT_CF_COUNT);
+
+        expectFirstValuesToBe(
+                stream,
+                expectedValues,
+                "HBase source did not produce the right values after a basic put operation");
+        doAndWaitForSuccess(
+                env, () -> cluster.put(baseTableName, DEFAULT_CF_COUNT, expectedValues), 120);
+    }
+
+    @Test
+    public void testOnlyReplicateSpecifiedTable() throws Exception {
+        String secondTable = baseTableName + "-table2";
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        DataStream<String> stream = streamFromHBaseSource(env, baseTableName);
+        cluster.makeTable(baseTableName, DEFAULT_CF_COUNT);
+        cluster.makeTable(secondTable, DEFAULT_CF_COUNT);
+        String[] expectedValues = uniqueValues(DEFAULT_CF_COUNT);
+
+        expectFirstValuesToBe(
+                stream,
+                expectedValues,
+                "HBase source did not produce the values of the correct table");
+        doAndWaitForSuccess(
+                env,
+                () -> {
+                    cluster.put(secondTable, DEFAULT_CF_COUNT, uniqueValues(DEFAULT_CF_COUNT));
+                    try {
+                        Thread.sleep(2000);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    cluster.put(baseTableName, DEFAULT_CF_COUNT, expectedValues);
+                },
+                180);
+    }
+
+    @Test
+    public void testRecordsAreProducedExactlyOnceWithCheckpoints() throws Exception {
+        final String collectedValueSignal = "collectedValue";
+        String[] expectedValues = uniqueValues(20);
+        cluster.makeTable(baseTableName);
+
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.enableCheckpointing(2000);
+        env.getCheckpointConfig().setCheckpointingMode(CheckpointingMode.EXACTLY_ONCE);
+        DataStream<String> stream = streamFromHBaseSource(env, baseTableName);
+        stream.addSink(
+                new FailureSink<String>(true, 3500, TypeInformation.of(String.class)) {
+
+                    private void checkForSuccess() {
+                        List<String> checkpointed = getCheckpointedValues();
+                        System.out.println(unCheckpointedValues + " " + checkpointed);
+                        if (checkpointed.size() == expectedValues.length) {
+                            try {
+                                assertArrayEquals(
+                                        "Wrong values were produced.",
+                                        expectedValues,
+                                        checkpointed.toArray());
+                                signalSuccess();
+                            } catch (Exception e) {
```

Comment:
```
catch (Exception e) {
    signalFailure();
    throw e;
}
```

Reply:
Done in 541e40635914f4c7fac89789c5dc9cd13455e6e5